rasa/core/test.py

```diff
@@ -719,19 +736,20 @@ async def _predict_tracker_actions(
             if entity_result:
                 policy_entity_results.append(entity_result)
 
-            tracker_eval_store.merge_store(action_executed_result)
-            tracker_actions.append(
-                {
-                    "action": action_executed_result.action_targets[0],
-                    "predicted": action_executed_result.action_predictions[0],
-                    "policy": prediction.policy_name,
-                    "confidence": prediction.max_confidence,
-                }
-            )
-            should_predict_another_action = processor.should_predict_another_action(
-                action_executed_result.action_predictions[0]
-            )
-            num_predicted_actions += 1
+            if action_executed_result.action_targets:
```

Comment:
Why is there a need for a `if` condition here?

Reply:
Because now the `action_unlikely_intent` will not be added to `action_executed_result.action_targets` and it means getting `action_executed_result.action_targets[0]` might be problematic